File: markdowner-filter/src/test/java/org/jesperancinha/parser/markdowner/filter/MavenFilterTest.java
package org.jesperancinha.parser.markdowner.filter;

import org.junit.jupiter.api.Test;

import java.net.URL;
import java.nio.file.Path;

import static org.assertj.core.api.Assertions.assertThat;

public class MavenFilterTest {

    @Test
    public void testMavenFilter() {
        final var resource1 = getClass().getResource("/directory2NoReadme/project1Maven/pom.xml");
        assertThat(resource1).isNotNull();
        final var resource = Path.of(resource1.getPath());
        final var mavenFilter = new MavenFilter();

        boolean result = mavenFilter.test(resource);
        assertThat(result).isTrue();
        assertThat(mavenFilter.lastProjectName()).isEqualTo("This is a test project");
    }

    @Test
    public void testMavenFilterFail() {
        final var resource1 = getClass().getResource("/directory2NoReadme/project3MavenAndNPM/package.json");
        assertThat(resource1).isNotNull();
        final var resource = Path.of(resource1.getPath());
        final var mavenFilter = new MavenFilter();

        final var result = mavenFilter.test(resource);

        assertThat(result).isFalse();
    }
}

File: markdowner-filter/src/test/kotlin/org/jesperancinha/parser/markdowner/filter/MavenFilterTest.kt
package org.jesperancinha.parser.markdowner.filter

import io.kotest.matchers.booleans.shouldBeFalse
import io.kotest.matchers.booleans.shouldBeTrue
import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import org.junit.jupiter.api.Test
import java.nio.file.Path

class MavenFilterTest {
    @Test
    fun testMavenFilter() {
        val pomXml = javaClass.getResource("/directory2NoReadme/project1Maven/pom.xml")
        pomXml.shouldNotBeNull()
        val resource = Path.of(pomXml.path)
        val mavenFilter = MavenFilter()
        val result = mavenFilter.test(resource)
        result.shouldBeTrue()
        mavenFilter.lastProjectName() shouldBe "This is a test project"
    }

    @Test
    fun testMavenFilterFail() {
        val packageJson = javaClass.getResource("/directory2NoReadme/project3MavenAndNPM/package.json")
        packageJson.shouldNotBeNull()
        val resource = Path.of(packageJson.path)
        val mavenFilter = MavenFilter()
        val result = mavenFilter.test(resource)
        result.shouldBeFalse()
    }
}